### DISCUSSION TITLE
How to access generated types in nuxt plugin?

### DISCUSSION THREAD

**Discussion by blasdfaa** [nuxt/nuxt#25804] [Questions]
Hi!

I'm building a nuxt module with type generation. In a module, I want to generate types, add them to `.nuxt/types/`, and then use them in the plugin to pass them to the generic function. How can i do this?

I want something like this:
```ts
// module.ts

export default defineNuxtModule({
  meta: {
   // ...
  },
  defaults: {
   // ...
  },
  setup(_options, nuxt) {
    addTypeTemplate({
      filename: 'foo-types.d.ts',
      getContents: () => `type TypeForGeneric = Record<string, unknown>`,
    })

    nuxt.hook('prepare:types', async ({ references }) => {
      references.push({ path: resolve(nuxt.options.buildDir, 'types/foo-types.d.ts') })
    })
    
    addPlugin(resolve('./runtime/plugin'))
  },
})

```
```ts
// runtime/plugin.ts

import { defineNuxtPlugin } from '#app'
import type { TypeForGeneric } from '' // <= where can I import it from?

export default defineNuxtPlugin((nuxtApp) => {
  const fooWithTypes = foo<TypeForGeneric>()

  return {
    provide: {
      fooWithTypes,
    },
  }
})
```

**Comment by xjccc** [comment]
`#build/foo-types.d.ts`?

**Reply by blasdfaa** [reply]
i get `Cannot find module '#build/foo-types.d.ts' or its corresponding type declarations.` in `runtime/plugin.ts`. 

But in the app when I use this module I can get these types from this path (file - `playground/plugins/foo.ts`. And I need to get this type in the plugin at the module level

**Reply by xjccc** [reply]
oh, i'm sorry that. i said it `addTemplate` in `#build`.
i think if declare d.ts, it will in global types?

**Reply by xjccc** [reply]
i think you can use `addTemplate` to generate a file to `export` your type?

**Reply by blasdfaa** [reply]
Yes, it will work. This was my very first solution.

But can I avoid creating a file in the working directory when using this module?

**Reply by xjccc** [reply]
`declare type TypeForGeneric = Record<string, unknown>` as global?

**Reply by blasdfaa** [reply]
This will only work for module users. But how do I get this type when developing a module?

**Reply by xjccc** [reply]
I think you can create a file in your project. And import it in your plugin.ts
